tests/benchmark/bm_basics.cpp

```diff
@@ -27,122 +27,166 @@ static void GetHNSWIndex(VecSimIndex *hnsw_index) {
     }
 }
 
+static void GetTestVectors(std::vector<std::vector<float>> &queries, size_t n_queries, size_t dim) {
+    auto location = std::string(std::string(getenv("ROOT")));
+    auto file_name = location + "/tests/benchmark/data/DBpedia-test_vectors-n10k.raw";
+
+    std::ifstream input(file_name, std::ios::binary);
+
+    queries.reserve(n_queries);
+    if (input.is_open()) {
+        input.seekg(0, std::ifstream::beg);
+        for (size_t i = 0; i < n_queries; i++) {
+            std::vector<float> query(dim);
+            input.read((char *)query.data(), dim * sizeof(float));
+            queries[i] = query;
+        }
+    } else {
+        std::cerr << "Test vectors file was not found in path. Exiting..." << std::endl;
+        exit(1);
+    }
+}
+
 class BM_VecSimBasics : public benchmark::Fixture {
 protected:
-    std::mt19937 rng;
     VecSimIndex *bf_index;
     VecSimIndex *hnsw_index;
     size_t dim;
     size_t n_vectors;
-    std::vector<float> query;
+    std::vector<std::vector<float>> *queries;
```

Comment:
Must be a pointer?

Reply:
To avoid copying it when we create several instances